src/Integrations/IntegrationRegistry.php

```diff
@@ -0,0 +1,179 @@
+
+		/**
+		 * Hook: integration_registration.
+		 *
+		 * Runs before integrations are initialized allowing new integration to be registered for use. This should be
+		 * used as the primary hook for integrations to include their scripts, styles, and other code extending the
+		 * blocks.
+		 *
+		 * @param IntegrationRegistry $this Instance of the IntegrationRegistry class which exposes the IntegrationRegistry::register() method.
+		 */
+		do_action( 'woocommerce_blocks_' . $this->registry_identifier . '_registration', $this );
+
+		foreach ( $this->get_all_registered() as $registered_integration ) {
+			$registered_integration->initialize();
+		}
+	}
+
+	/**
+	 * Registers an integration.
+	 *
+	 * @param IntegrationInterface $integration An instance of IntegrationInterface.
+	 *
+	 * @return boolean True means registered successfully.
+	 */
+	public function register( IntegrationInterface $integration ) {
+		$name = $integration->get_name();
+
+		if ( $this->is_registered( $name ) ) {
+			/* translators: %s: Class name. */
+			_doing_it_wrong( __METHOD__, esc_html( sprintf( __( '"%s" is already registered.', 'woo-gutenberg-products-block' ), $name ) ) );
+			return false;
+		}
+
+		$this->registered_integrations[ $name ] = $integration;
+		return true;
+	}
+
+	/**
+	 * Checks if an integration is already registered.
+	 *
+	 * @param string $name Integration name.
+	 * @return bool True if the integration is registered, false otherwise.
+	 */
+	public function is_registered( $name ) {
+		return isset( $this->registered_integrations[ $name ] );
+	}
+
+	/**
+	 * Un-register an integration.
+	 *
+	 * @param string|IntegrationInterface $name Integration name, or alternatively a IntegrationInterface instance.
+	 * @return boolean|IntegrationInterface Returns the unregistered integration instance if unregistered successfully.
+	 */
+	public function unregister( $name ) {
+		if ( $name instanceof IntegrationInterface ) {
+			$name = $name->get_name();
+		}
+
+		if ( ! $this->is_registered( $name ) ) {
+			/* translators: %s: Integration name. */
+			_doing_it_wrong( __METHOD__, esc_html( sprintf( __( 'Integration "%s" is not registered.', 'woo-gutenberg-products-block' ), $name ) ) );
+			return false;
+		}
+
+		$unregistered = $this->registered_integrations[ $name ];
+		unset( $this->registered_integrations[ $name ] );
+		return $unregistered;
+	}
+
+	/**
+	 * Retrieves a registered Integration by name.
+	 *
+	 * @param string $name Integration name.
+	 * @return IntegrationInterface|null The registered integration, or null if it is not registered.
+	 */
+	public function get_registered( $name ) {
+		return $this->is_registered( $name ) ? $this->registered_integrations[ $name ] : null;
+	}
+
+	/**
+	 * Retrieves all registered integrations.
+	 *
+	 * @return IntegrationInterface[]
+	 */
+	public function get_all_registered() {
+		return $this->registered_integrations;
+	}
+
+	/**
+	 * Gets an array of all registered integration's script handles.
```

Comment:
The comment needs `editor` word somewhere otherwise this is the same thing as for `get_all_registered_script_handles`